{
  "SchemaVersion": 2,
  "ArtifactName": ".",
  "ArtifactType": "filesystem",
  "Metadata": {
    "ImageConfig": {
      "architecture": "",
      "created": "0001-01-01T00:00:00Z",
      "os": "",
      "rootfs": {
        "type": "",
        "diff_ids": null
      },
      "config": {}
    }
  },
  "Results": [
    {
      "Target": "poetry.lock",
      "Class": "lang-pkgs",
      "Type": "poetry",
      "Vulnerabilities": [
        {
          "VulnerabilityID": "CVE-2024-39689",
          "PkgName": "certifi",
          "InstalledVersion": "2024.2.2",
          "FixedVersion": "2024.07.04",
          "Layer": {},
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-39689",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "python-certifi: Remove root certificates from `GLOBALTRUST` from the root store",
          "Description": "Certifi is a curated collection of Root Certificates for validating the trustworthiness of SSL certificates while verifying the identity of TLS hosts. Certifi starting in 2021.05.30 and prior to 2024.07.4 recognized root certificates from `GLOBALTRUST`. Certifi 2024.07.04 removes root certificates from `GLOBALTRUST` from the root store. These are in the process of being removed from Mozilla's trust store. `GLOBALTRUST`'s root certificates are being removed pursuant to an investigation which identified \"long-running and unresolved compliance issues.\"",
          "Severity": "LOW",
          "CweIDs": [
            "CWE-345"
          ],
          "CVSS": {
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:L/I:N/A:N",
              "V3Score": 3.7
            }
          },
          "References": [
            "https://access.redhat.com/security/cve/CVE-2024-39689",
            "https://github.com/certifi/python-certifi",
            "https://github.com/certifi/python-certifi/commit/bd8153872e9c6fc98f4023df9c2deaffea2fa463",
            "https://github.com/certifi/python-certifi/security/advisories/GHSA-248v-346w-9cwc",
            "https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/XpknYMPO8dI",
            "https://nvd.nist.gov/vuln/detail/CVE-2024-39689",
            "https://www.cve.org/CVERecord?id=CVE-2024-39689"
          ],
          "PublishedDate": "2024-07-05T19:15:10.247Z",
          "LastModifiedDate": "2024-07-08T15:49:22.437Z"
        },
        {
          "VulnerabilityID": "GHSA-h4gh-qq45-vh27",
          "PkgName": "cryptography",
          "InstalledVersion": "42.0.5",
          "FixedVersion": "43.0.1",
          "Layer": {},
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://github.com/advisories/GHSA-h4gh-qq45-vh27",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "pyca/cryptography has a vulnerable OpenSSL included in cryptography wheels",
          "Description": "pyca/cryptography's wheels include a statically linked copy of OpenSSL. The versions of OpenSSL included in cryptography 37.0.0-43.0.0 are vulnerable to a security issue. More details about the vulnerability itself can be found in https://openssl-library.org/news/secadv/20240903.txt.\n\nIf you are building cryptography source (\"sdist\") then you are responsible for upgrading your copy of OpenSSL. Only users installing from wheels built by the cryptography project (i.e., those distributed on PyPI) need to update their cryptography versions.\n",
          "Severity": "MEDIUM",
          "References": [
            "https://github.com/pyca/cryptography",
            "https://github.com/pyca/cryptography/security/advisories/GHSA-h4gh-qq45-vh27",
            "https://openssl-library.org/news/secadv/20240903.txt"
          ]
        },
        {
          "VulnerabilityID": "CVE-2024-35195",
          "PkgName": "requests",
          "InstalledVersion": "2.31.0",
          "FixedVersion": "2.32.0",
          "Layer": {},
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-35195",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "requests: subsequent requests to the same host ignore cert verification",
          "Description": "Requests is a HTTP library. Prior to 2.32.0, when making requests through a Requests `Session`, if the first request is made with `verify=False` to disable cert verification, all subsequent requests to the same host will continue to ignore cert verification regardless of changes to the value of `verify`. This behavior will continue for the lifecycle of the connection in the connection pool. This vulnerability is fixed in 2.32.0.",
          "Severity": "MEDIUM",
          "CweIDs": [
            "CWE-670"
          ],
          "CVSS": {
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:L/AC:H/PR:H/UI:R/S:U/C:H/I:H/A:N",
              "V3Score": 5.6
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:L/AC:H/PR:H/UI:R/S:U/C:H/I:H/A:N",
              "V3Score": 5.6
            }
          },
          "References": [
            "https://access.redhat.com/security/cve/CVE-2024-35195",
            "https://github.com/psf/requests",
            "https://github.com/psf/requests/commit/a58d7f2ffb4d00b46dca2d70a3932a0b37e22fac",
            "https://github.com/psf/requests/pull/6655",
            "https://github.com/psf/requests/security/advisories/GHSA-9wx4-h78v-vm56",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/IYLSNK5TL46Q6XPRVMHVWS63MVJQOK4Q",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/IYLSNK5TL46Q6XPRVMHVWS63MVJQOK4Q/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/N7WP6EYDSUOCOJYHDK5NX43PYZ4SNHGZ",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/N7WP6EYDSUOCOJYHDK5NX43PYZ4SNHGZ/",
            "https://nvd.nist.gov/vuln/detail/CVE-2024-35195",
            "https://www.cve.org/CVERecord?id=CVE-2024-35195"
          ],
          "PublishedDate": "2024-05-20T21:15:09.99Z",
          "LastModifiedDate": "2024-06-10T17:16:29.563Z"
        },
        {
          "VulnerabilityID": "CVE-2024-6345",
          "PkgName": "setuptools",
          "InstalledVersion": "69.5.1",
          "FixedVersion": "70.0.0",
          "Layer": {},
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-6345",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "pypa/setuptools: Remote code execution via download functions in the package_index module in pypa/setuptools",
          "Description": "A vulnerability in the package_index module of pypa/setuptools versions up to 69.1.1 allows for remote code execution via its download functions. These functions, which are used to download packages from URLs provided by users or retrieved from package index servers, are susceptible to code injection. If these functions are exposed to user-controlled inputs, such as package URLs, they can execute arbitrary commands on the system. The issue is fixed in version 70.0.",
          "Severity": "HIGH",
          "CweIDs": [
            "CWE-94"
          ],
          "CVSS": {
            "bitnami": {
              "V3Vector": "CVSS:3.0/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
              "V3Score": 8.8
            },
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
              "V3Score": 8.8
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
              "V3Score": 8.8
            }
          },
          "References": [
            "https://access.redhat.com/errata/RHSA-2024:5534",
            "https://access.redhat.com/security/cve/CVE-2024-6345",
            "https://bugzilla.redhat.com/2297771",
            "https://bugzilla.redhat.com/show_bug.cgi?id=2297771",
            "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-6345",
            "https://errata.almalinux.org/9/ALSA-2024-5534.html",
            "https://errata.rockylinux.org/RLSA-2024:5530",
            "https://github.com/pypa/setuptools",
            "https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0",
            "https://github.com/pypa/setuptools/pull/4332",
            "https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5",
            "https://linux.oracle.com/cve/CVE-2024-6345.html",
            "https://linux.oracle.com/errata/ELSA-2024-6311.html",
            "https://nvd.nist.gov/vuln/detail/CVE-2024-6345",
            "https://www.cve.org/CVERecord?id=CVE-2024-6345"
          ],
          "PublishedDate": "2024-07-15T01:15:01.73Z",
          "LastModifiedDate": "2024-07-15T13:00:34.853Z"
        },
        {
          "VulnerabilityID": "CVE-2024-37891",
          "PkgName": "urllib3",
          "InstalledVersion": "2.2.1",
          "FixedVersion": "1.26.19, 2.2.2",
          "Layer": {},
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-37891",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "urllib3: proxy-authorization request header is not stripped during cross-origin redirects",
          "Description": " urllib3 is a user-friendly HTTP client library for Python. When using urllib3's proxy support with `ProxyManager`, the `Proxy-Authorization` header is only sent to the configured proxy, as expected. However, when sending HTTP requests *without* using urllib3's proxy support, it's possible to accidentally configure the `Proxy-Authorization` header even though it won't have any effect as the request is not using a forwarding proxy or a tunneling proxy. In those cases, urllib3 doesn't treat the `Proxy-Authorization` HTTP header as one carrying authentication material and thus doesn't strip the header on cross-origin redirects. Because this is a highly unlikely scenario, we believe the severity of this vulnerability is low for almost all users. Out of an abundance of caution urllib3 will automatically strip the `Proxy-Authorization` header during cross-origin redirects to avoid the small chance that users are doing this on accident. Users should use urllib3's proxy support or disable automatic redirects to achieve safe processing of the `Proxy-Authorization` header, but we still decided to strip the header by default in order to further protect users who aren't using the correct approach. We believe the number of usages affected by this advisory is low. It requires all of the following to be true to be exploited: 1. Setting the `Proxy-Authorization` header without using urllib3's built-in proxy support. 2. Not disabling HTTP redirects. 3. Either not using an HTTPS origin server or for the proxy or target origin to redirect to a malicious origin. Users are advised to update to either version 1.26.19 or version 2.2.2. Users unable to upgrade may use the `Proxy-Authorization` header with urllib3's `ProxyManager`, disable HTTP redirects using `redirects=False` when sending requests, or not user the `Proxy-Authorization` header as mitigations.",
          "Severity": "MEDIUM",
          "CweIDs": [
            "CWE-669"
          ],
          "CVSS": {
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N",
              "V3Score": 4.4
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N",
              "V3Score": 4.4
            }
          },
          "References": [
            "https://access.redhat.com/errata/RHSA-2024:6162",
            "https://access.redhat.com/security/cve/CVE-2024-37891",
            "https://bugzilla.redhat.com/2292788",
            "https://errata.almalinux.org/9/ALSA-2024-6162.html",
            "https://github.com/urllib3/urllib3",
            "https://github.com/urllib3/urllib3/commit/40b6d1605814dd1db0a46e202d6e56f2e4c9a468",
            "https://github.com/urllib3/urllib3/commit/accff72ecc2f6cf5a76d9570198a93ac7c90270e",
            "https://github.com/urllib3/urllib3/security/advisories/GHSA-34jh-p97f-mpxf",
            "https://linux.oracle.com/cve/CVE-2024-37891.html",
            "https://linux.oracle.com/errata/ELSA-2024-6311.html",
            "https://nvd.nist.gov/vuln/detail/CVE-2024-37891",
            "https://www.cve.org/CVERecord?id=CVE-2024-37891"
          ],
          "PublishedDate": "2024-06-17T20:15:13.45Z",
          "LastModifiedDate": "2024-06-20T12:44:22.977Z"
        }
      ]
    }
  ]
}
